build.gradle

```diff
@@ -295,7 +296,12 @@ subprojects {
       }
       publications {
         mavenJava(MavenPublication) {
-          from components.java
+          if (!shouldPublishWithShadow) {
+            from components.java
+          } else {
+            apply plugin: 'com.github.johnrengelman.shadow'
```

Comment:
What is this plugin? Does not look like something official? Not sure if we should use something that is no backed by some official OSS project?

Reply:
Valid ask, we require this plugin to move Opentelemetry required libs as shadowed to avoid version conflicts when higher version of these commonly used libraries are present in user's application code, but this [link](https://imperceptiblethoughts.com/shadow/introduction/) for the plugin details better. This dependency is not introduced in this PR rather being already used in Kafka for a while: 

https://github.com/apache/kafka/blob/f0e97397c01241be9d8d05b1642352b58d1ed196/build.gradle#L47

Reply:
Thanks. Yes, I understood already what the plugin does.

> This dependency is not introduced in this PR rather being already used in Kafka for a while:

This was my concern, but if it's already established my concerns are void.